### PR TITLE
Remove six and pathlib2 dependencies

### PR DESCRIPTION
**Pull request by ssbarnea**
As we removed support for py27, we can now drop two dependencies that where used in order to provide support for it.

Partial-Fix: #699

### CODE REVIEW

File: docs/docsite/rst/conf.py [hunk 1]
```diff
@@ -228,7 +228,7 @@
 
 autoclass_content = 'both'
 
-intersphinx_mapping = {'python': ('https://docs.python.org/2/', (None, '../python2-2.7.13.inv')),
+intersphinx_mapping = {'python': ('https://docs.python.org/3/', (None, '../python3-3.6.2.inv')),
```
Comment (webknjaz) [inline docs/docsite/rst/conf.py:231]: I'm not sure if we should do this now. The docs may as well still reference things from py2. Let's omit this.

File: lib/ansiblelint/__init__.py [hunk 2]
```diff
@@ -66,7 +64,7 @@ def matchlines(self, file, text):
             if not result:
                 continue
             message = None
-            if isinstance(result, six.string_types):
+            if isinstance(result, str):
```
Comment (webknjaz) [inline lib/ansiblelint/__init__.py:67]: Let's revert all the modifications to `six.string_types` for now as what you're doing effectively modifies logic. Before all checks were "bytes or unicode" not it's just "unicode" which may lead to unforeseen behavior changes. So I want to see less of dangerous changes in this PR.

File: lib/ansiblelint/utils.py [hunk 3]
```diff
@@ -23,13 +23,10 @@
 import imp
 from itertools import product
 import os
-try:
-    from pathlib import Path
-except ImportError:
-    from pathlib2 import Path
+from pathlib import Path
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py:26]: LGTM

File: zuul.d/layout.yaml [hunk 4]
```diff
@@ -1,40 +0,0 @@
----
```
Comment (webknjaz) [inline zuul.d/layout.yaml:1]: What does this have to do with Python 2 removal?

File: lib/ansiblelint/__init__.py (hunk 2 above)
Comment (ssbarnea) [reply lib/ansiblelint/__init__.py:67]: I really doubt there is any change in behavior, take a look at https://six.readthedocs.io/#six.string_types -- I did this switch on other projects.

As long we run on python3 there is no "unicode" anymore. The only case where it would be a change in behavior is if we endup running on py2, which we do not allow or support.

File: zuul.d/layout.yaml (hunk 4 above)
Comment (ssbarnea) [reply zuul.d/layout.yaml:1]: In fact the entire zuul.d/ needs to removed as is not used by zuul.

File: lib/ansiblelint/__init__.py (hunk 2 above)
Comment (webknjaz) [reply lib/ansiblelint/__init__.py:67]: Okay, keep it then.

File: zuul.d/layout.yaml (hunk 4 above)
Comment (webknjaz) [reply zuul.d/layout.yaml:1]: Remove it separately then

**Comment by ssbarnea** [comment]
I will stage it, first step is at https://github.com/ansible/ansible-lint/pull/715 which removes the jobs. I will also do the other changes but separated by subject, making easier to review.

**Comment by webknjaz** [comment]
Great, thanks! This usually makes time to merge smaller for non-controversial changes in general and fixing one thing doesn't affect merging other things.
